TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/CANFilterService.java

```diff
@@ -0,0 +1,216 @@
+ *  (C) Copyright Testing Technologies.  All Rights Reserved.
+ *
+ *  All copies of this program, whether in whole or in part, and whether
+ *  modified or not, must display this and all other embedded copyright
+ *  and ownership notices in full.
+ *
+ *  See the file COPYRIGHT for details of redistribution and use.
+ *
+ *  You should have received a copy of the COPYRIGHT file along with
+ *  this file; if not, write to the Testing Technologies,
+ *  Michaelkirchstr. 17/18, 10179 Berlin, Germany.
+ *
+ *  TESTING TECHNOLOGIES DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS
+ *  SOFTWARE. IN NO EVENT SHALL TESTING TECHNOLOGIES BE LIABLE FOR ANY
+ *  SPECIAL, DIRECT, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+ *  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ *  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+ *  THIS SOFTWARE.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ *  EITHER EXPRESSED OR IMPLIED, INCLUDING ANY KIND OF IMPLIED OR
+ *  EXPRESSED WARRANTY OF NON-INFRINGEMENT OR THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.
+ * ----------------------------------------------------------------------------
+ */
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+import java.lang.Thread.State;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.Hashtable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class CANFilterService {
+
+  private static final int DEFAULT_SERVER_PORT = 7000;
+  private static final String DEFAULT_SERVER_HOST = "localhost";
+
+  private static Hashtable<String, Car2XEntry> car2xEntries = new Hashtable<>();
+  private static Car2XEntryUpdater entryUpdater;
+  private static TimeoutResponder timeoutResponder;
+
+  /**
+   * Provides tcp server functionality to handle json requests. Upon receiving 
+   * a start request, an entry updater thread which updates openXC / obd2 hash 
+   * table entries is started. Parallel timeout responder tasks periodically
+   * send back updated JSON values from the hash table over a client socket.
+   * 
+   * @param args
+   *            host, port
+   */
+  public static void main(String[] args) {
+    String host = DEFAULT_SERVER_HOST;
+    int port = DEFAULT_SERVER_PORT;
+
+    if (args.length != 2) {
+      System.err
+          .println("[CANFilterService] Usage: java CANFilterService "
+              + "<Server host>  <Server Port Number>");
+      System.err.println("  ... using default server host "
+          + DEFAULT_SERVER_HOST + " default server port "
+          + DEFAULT_SERVER_PORT);
+    } else {
+      host = args[0];
+      port = extractPortNumber(args, 1, DEFAULT_SERVER_PORT);
```

Comment:
Please use "this", when you call methods, which are implemented in the same class.